scripts/dqlite/cmd/main.go

```diff
@@ -171,23 +172,32 @@ func main() {
 				return
 			}
 
-			// Process the input query.
-			result, err := processQuery(ctx, db, input)
-			if err != nil {
-				fmt.Fprintln(os.Stderr, "Error: ", err)
-			} else {
-				line.AppendHistory(input)
-				if result != "" {
-					fmt.Println()
-					fmt.Println(result)
-				}
+			lineBuildUp.WriteString(input + " ")
+			if !strings.HasSuffix(strings.TrimSpace(input), ";") {
+				continue
+			}
 
-				// This can fail, so we don't care if it errors out.
-				if file != nil {
-					fmt.Fprintln(file, input)
-					_ = file.Sync()
+			func() {
+				defer lineBuildUp.Reset()
+
+				// Process the input query.
+				result, err := processQuery(ctx, db, lineBuildUp.String())
+				if err != nil {
+					fmt.Fprintln(os.Stderr, "Error: ", err)
+				} else {
+					line.AppendHistory(lineBuildUp.String())
```

Comment:
Whilst you're here, just always add the append to history. It's really good if you make a mistake to be able to backscroll up to edit the mistake. Currently the query has to succeed if you want history, which is terrible UX.

Reply:
Done & Done